components/sst_importer/src/import_file.rs

```diff
@@ -458,20 +458,33 @@ impl ImportDir {
 }
 
 const SST_SUFFIX: &str = ".sst";
-
+// version 2: compared to version 1 which is the default version, we will check
+// epoch of request and local region in write API.
+pub const API_VERSION_2: i32 = 2;
+
+/// sst_meta_to_path will encode the filepath with default api version (current
+/// is 2). So when the SstMeta is created in old version of TiKV and filepath
+/// will not correspond to the real file, in the deletion logic we can't remove
+/// these files.
 pub fn sst_meta_to_path(meta: &SstMeta) -> Result<PathBuf> {
     Ok(PathBuf::from(format!(
-        "{}_{}_{}_{}_{}{}",
+        "{}_{}_{}_{}_{}_{}{}",
         UuidBuilder::from_slice(meta.get_uuid())?.build(),
         meta.get_region_id(),
         meta.get_region_epoch().get_conf_ver(),
         meta.get_region_epoch().get_version(),
         meta.get_cf_name(),
+        API_VERSION_2,
         SST_SUFFIX,
     )))
 }
 
-pub fn parse_meta_from_path<P: AsRef<Path>>(path: P) -> Result<SstMeta> {
+pub struct SstMetaWithApiVersion {
```

Comment:
The naming of this struct suggests that it's unncessary to define the struct at all. Just (SstMeta，api_version) is good.

Reply:
https://github.com/tikv/tikv/pull/15795/commits/da8ab023d84d876ffd8e1e766484e4250ddb67df